### PR TITLE
Plumb through a structured key, keep current behavior.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, this is an attempt to introduce a structured key to be used by our reconcilers to remove unnecessary boilerplate.

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -306,7 +313,8 @@ func (c *Impl) processNextWorkItem() bool {
 	if shutdown {
 		return false
 	}
-	key := obj.(string)
+	key := obj.(types.NamespacedName)
+	fqdn := key.String()
```
Comment (mattmoor) [inline controller/controller.go]: Fully Qualified Domain Name?

Comment (markusthoemmes) [reply controller/controller.go]: /facepalm

**Comment by markusthoemmes** [comment]
@mattmoor thanks, ready for another look.

**Pull request merged**